ccm_web/server/src/lti/lti.service.ts

```diff
@@ -66,6 +67,17 @@ export class LTIService implements BeforeApplicationShutdown {
       const roles = customLTIVariables.roles as string
       const isRootAdmin = customLTIVariables.is_root_account_admin as boolean
 
+      // check whether the user has at least one of the allowed LTI user roles
+      // otherwise block tool access and show error message
+      const rolesLTI = roles.length > 0 ? roles.split(',') : []
+      const ltiAllowedRoles = rolesLTI.filter(
+        x => Object.entries(LTIEnrollmentTypes).find(
+          ([key, value]) => value === x)
+      )
+      if (ltiAllowedRoles.length === 0) {
+        return createLaunchErrorResponse(res, 'Your role in this course does not allow access to this tool. If you feel this is in error, please contact 4help@umich.edu.')
```

Comment:
Just suggesting you lowercase "your" here the text before ends with a semicolon.
```suggestion
        return createLaunchErrorResponse(res, 'your role in this course does not allow access to this tool. If you feel this is in error, please contact 4help@umich.edu.')
```

Reply:
text updated.